eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -460,10 +472,19 @@ def group_variables(model):
 
     - December 29, 2021 Taylor Bell
         Moved code to separate function to reduce repeated code.
+    - January 11, 2022 Megan Mansfield
+        Added ability to have shared parameters
     """
     # all_params = [i for j in [model.components[n].parameters.dict.items()
     #               for n in range(len(model.components))] for i in j]
-    all_params = list(model.components[0].parameters.dict.items())
+    if model.components[0].share:
```

Comment:
Can you explain the logic behind this chunk of code and maybe add some comments to the code itself as well? I'm having a hard time figuring out the function and nuance of the code

Reply:
Yeah, so the logic here is basically that for a shared fit you want the list of free parameters to include all the parameters for the individual visits (e.g., rp_0, rp_1, rp_2...t0_0, t0_1, t0_2...), but for a fit of a single channel the free parameters should only include that channel's parameter (rp_1, t0_1, etc.). This can also be simplified using your nchan=1 for a non-shared fit, and I'll also add comments to clarify what's going on.

Reply:
In my opinion, the latter is greatly preferred to reduce redundant labour. I can't really think of a time where one would want to enforce separate priors on each different wavelength's parameters, but @kevin218 might have a different opinion

Reply:
Actually I got rid of this bit of the code because it's unnecessary with the new nchannels=1 thing. And to answer your question, the user just has to put in one value of rp and somewhere in the code it duplicates that. Perhaps that is a thing we will want to change at some point if we want to give the user the option to specify individual values for each channel, but for now it's working fine to do it that way and it makes it much less messy on the user end.